modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -272,60 +396,36 @@ impl WalletClientModule {
     }
 
     pub fn get_network(&self) -> Network {
-        self.cfg.network
+        self.cfg().network
     }
 
     pub fn get_fee_consensus(&self) -> FeeConsensus {
-        self.cfg.fee_consensus
+        self.cfg().fee_consensus
     }
 
     pub async fn get_deposit_address_inner(
         &self,
-        valid_until: SystemTime,
         dbtx: &mut DatabaseTransaction<'_>,
-    ) -> (OperationId, WalletClientStates, Address) {
+    ) -> (OperationId, Address, TweakIdx) {
         let deposit_idx = get_next_peg_in_tweak_child_id(dbtx).await;
```

Comment:
[nit] renaming `deposit_idx` to `tweak_idx` would make this consistent with other usages